lib/msf/core/auxiliary/vyos.rb

```diff
@@ -0,0 +1,306 @@
+# -*- coding: binary -*-
+
+require 'metasploit/framework/hashes/identify'
+
+module Msf
+  ###
+  #
+  # This module provides methods for working with VyOS equipment
+  #
+  ###
+  module Auxiliary::VYOS
+    include Msf::Auxiliary::Report
+
+    def vyos_config_eater(thost, tport, config, store = true)
+
+      credential_data = {
+        address: thost,
+        port: tport,
+        protocol: 'tcp',
+        workspace_id: myworkspace.id,
+        origin_type: :service,
+        private_type: :nonreplayable_hash,
+        # jtr_format: 'sha512,crypt', # default on the devices 11.4.0+
+        service_name: '',
+        module_fullname: fullname,
+        status: Metasploit::Model::Login::Status::UNTRIED
+      }
+
+      # Default SNMP to UDP
+      if tport == 161
+        credential_data[:protocol] = 'udp'
+      end
+
+      if store && !config.include?('such file or directory') && !config.include?('ermission denied')
+        l = store_loot('vyos.config', 'text/plain', thost, config.strip, 'config.txt', 'VyOS Configuration')
+        vprint_good("#{thost}:#{tport} Config saved to: #{l}")
+      end
+
+      host_info = {
+        host: thost,
+        os_name: 'VyOS'
+      }
+      report_host(host_info)
+
+      # generated by: cat /config/config.boot
+      # https://github.com/rapid7/metasploit-framework/issues/14124
+
+      # login {
+      #    user jsmith {
+      #        authentication {
+      #            encrypted-password $6$ELBrDuW7c/8$nN7MwUST8s8O0R6HMNu/iPoTQ1s..y8HTnXraJ7Hh4bHefRmjt/2U08ZckEw4FU034wbWaeCaB5hq7mC6fNXl/
+      #            plaintext-password ""
+      #        }
+      #        full-name "John Smith"
+      #        level operator
+      #    }
+      #    user vyos {
+      #        authentication {
+      #            encrypted-password $1$5HsQse2v$VQLh5eeEp4ZzGmCG/PRBA1
+      #            plaintext-password ""
+      #        }
+      #        level admin
+      #    }
+      # }
+
+      # sometimes the hash is masked
+
+      # login {
+      #   user vyos {
+      #        authentication {
+      #            encrypted-password ****************
+      #            plaintext-password ""
+      #        }
+      #        level admin
+      #    }
+      # }
+
+      # plaintext-password can also be missing: https://github.com/rapid7/metasploit-framework/pull/14161#discussion_r492884039
+
+      r =  'user ([^ ]+) {\s*authentication {\s*'
+      r << 'encrypted-password (\$?[\w$\./\*]*)\s*' # leading $ is optional incase the password is all stars
+      r << '(?:plaintext-password "([^"]*)")?\s*' # optional
+      r << '}'
+      r << '(?:\s*full-name "([^"]*)")?\s*' # optional
+      r << 'level (operator|admin)' # 1.3+ seems to have removed operator
```

Comment:
It looks like `level` has been completely removed in version 1.3 and it breaks the regex:
```
    login {
        user jsmith {
            authentication {
                encrypted-password $6$BjDy4.nwPSY45GTX$V7vnDy774u4DVEXW8LxCr7VLD4KRaNkrpZlM9ojHdcs.x2MxFjyN4kMhofwWVdDqkd4546vSH5.jwiUzVmi0M/
                plaintext-password ""
            }
            full-name "John Smith"
        }
        user vyos {
            authentication {
                encrypted-password $6$QxPS.uk6mfo$9QBSo8u1FkH16gMyAVhus6fU3LOzvLR9Z9.82m3tiHFAxTtIkhaZSWssSgzt4v4dGAL8rhVQxTg0oAG9/q11h/
                plaintext-password ""
            }
        }
    }
```

Reply:
you are 100% correct. Strange to miss that, confirmed.

Reply:
also added use cases to ensure that doesn't get missed in the future!